### PR TITLE
Place the next pointer at a different place on every object.

### PR DESCRIPTION
**Pull request by plietar**
This limits the collision in cache buckets, especially for objects in the 512-8K range.

capture

This is behind a flag (CACHE_FRIENDLY_OFFSET) for the moment.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +48,10 @@ if(USE_SNMALLOC_STATS)
   add_definitions(-DUSE_SNMALLOC_STATS)
 endif()
 
+if(CACHE_FRIENDLY_OFFSET)
```
Comment (mjp41) [inline CMakeLists.txt]: Given the tests are setting this manually.  Should this just be added to the `add_shim` macro?

File: src/mem/sizeclass.h [hunk 2]
```diff
@@ -6,6 +6,9 @@ namespace snmalloc
 {
   constexpr static uint16_t get_slab_offset(uint8_t sc, bool is_short);
   constexpr static size_t sizeclass_to_size(uint8_t sizeclass);
+  constexpr static size_t sizeclass_to_cache_friendly_mask(uint8_t sizeclass);
+  constexpr static size_t
+  sizeclass_to_inverse_cache_friendly_mask(uint8_t sizeclass);
```
Comment (mjp41) [inline src/mem/sizeclass.h]: Is that really what clangformat gives here?

File: CMakeLists.txt (hunk 1 above)
Comment (plietar) [reply CMakeLists.txt]: Done

File: src/mem/sizeclass.h (hunk 2 above)
Comment (plietar) [reply src/mem/sizeclass.h]: Done

**Pull request merged**